core/arch/arm/plat-k3/drivers/ti_sci.h

```diff
@@ -161,6 +161,17 @@ int ti_sci_write_otp_row(uint8_t row_idx, uint32_t row_val, uint32_t row_mask);
 int ti_sci_lock_otp_row(uint8_t row_idx, uint8_t hw_write_lock,
 			uint8_t hw_read_lock, uint8_t row_soft_lock);
 
+/**
+ * ti_sci_set_swrev - Write Software Revision
+ * @identifier: One of the entries from enum tisci_otp_revision_identifier
+ * @swrev:	Software Revision
+ *
+ * Writes the software revision.
+ *
+ * Return: 0 if all goes well, else appropriate error message
+ */
+int ti_sci_set_swrev(uint8_t identifier, uint32_t swrev);
```

Comment:
Would it make sense to have `enum tisci_otp_revision_identifier` as type for `identifier`?
Maybe not if caller gets the value from a raw `int`, `uint8_t` or like.

Reply:
There will need to be a narrowing from `enum` to `uint8_t` at some point, I'd like to either make it the job of the caller or maybe have a couple versions of this function, one for each type of revision identifier, so we can hide away the enum from the API callers. This would also help to prevent calling with bad identifier enum values. Will think on that.